src/main/java/com/google/api/gax/testing/MockGrpcService.java

```diff
@@ -45,7 +45,10 @@
   public List<GeneratedMessageV3> getRequests();
 
   /** Sets the responses. */
-  public void setResponses(List<GeneratedMessageV3> responses);
+  public void setResponses(GeneratedMessageV3... responses);
+
+  /** Sets the exception. */
+  public void setExceptions(Exception... exceptions);
```

Comment:
When I was playing around with a manually-written instance of MockGrpcService, I found it much more powerful to have an interface like this:

```
public void addResponse(GeneratedMessageV3 response);
public void addException(Exception exception);
```

And then in the implementation, both cases would get added to a `LinkedList<Object>`; in the individual API calls, I would pull off the next object, and return a response or throw an exception appropriately. This allows for testing retry cases deterministically.